src/main/cpp/kernels/SystemML.ptx

```diff
@@ -0,0 +1,76 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
There are following options for supporting custom kernels:

1. Compile .ptx file on machine where nvcc is installed and include it as resource file (Current option). This will use CUDA driver-side API (similar to cublas calls) and has no dependency on whether nvcc + Visual Studio or g++.
**Pros:** Will work in distributed setting, no installation of SystemML libraries.
**Cons:** Will require .ptx to be committed to git.

2. Include .cu file as resource file and compile it during execution. Here are two options to do so:
a. Include jcuda utility into the classpath and use KernelLauncher: http://www.jcuda.org/utilities/KernelLauncherSample.java
b. Run nvcc: http://jcuda.org/samples/JCudaVectorAdd.java
**Pros:** Will not require .ptx to be committed to git.
**Cons:** GPU backend dependent on System setup. For example: nvcc 7.5 requires Visual Studio 10 on Windows.

4. Add gpu profile in pom.xml, which invokes nvcc during `mvn package`
**Pros:** Will not require .ptx to be committed to git.
**Cons:** Will increase the number of artifacts. 

5. Compile dll/so files and use JNI. 
**Cons:** Build complexity. Will require shipping and testing SystemML on different OS.

Including @lresende and @deroneriksson since this is related to build issues.